File: LeetCode Algorithms/125.验证回文串.py
#
# @lc app=leetcode.cn id=125 lang=python3
#
# [125] 验证回文串
#

# @lc code=start
class Solution:
    # def isPalindrome(self, s: str) -> bool:
    @classmethod
    def isPalindrome(self, s):
        # filter_str = lambda x: str.lower(''.join(filter(str.isalnum, str(x).replace(' ', ''))))
        # return filter_str(s) == ''.join(reversed(filter_str(s)))

        
# @lc code=end

print (Solution.isPalindrome(s = "1A man, a plan, a canal: Panama1"))

File: LeetCode Algorithms/125.验证回文串.cpp
/*
 * @lc app=leetcode.cn id=125 lang=cpp
 *
 * [125] 验证回文串
 */

// @lc code=start
class Solution {
public:
    bool isPalindrome(string s) {
        string tmp_s;
        for (char ch: s) {
            if (isalnum(ch)) {    // 判断字符是否为(大小写)字母或数字
                tmp_s += tolower(ch);
            }
        }

        int n = tmp_s.size();
        int left = 0, right = n - 1;
        while (left < right) {
            if (tmp_s[left] != tmp_s[right]) {
                return false;
            }
            ++left;
            --right;
        }
        return true;
    }
};
// @lc code=end

